ls1intum/artemis-android
# `Bugfix`: Removed moderator rights to edit posts; added test

### Problem Description
Currently a user with moderation rights can edit anyone's posts. As discussed in the Communication Team Meeting, this behaviour is unwanted.

### Changes
 
- Removed moderator rights to edit other's posts
- Created unit test


### Steps for testing
- Login as an instructor (has moderation rights)
- Long press another users post
- See that "Edit" is not available anymore
- Press on a post of yourself 
- "Edit" is still possible
